unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+    free(lp_iviShell->debug_scopes);
+    free(lp_iviShell);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_destroy_success
+ * @brief               Test case of controller_screen_destroy() where valid input params
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_destroy()
+ *                      -# Verification point:
+ *                         +# wl_resource_destroy() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_destroy_success)
+{
+    controller_screen_destroy(nullptr, nullptr);
+    ASSERT_EQ(wl_resource_destroy_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_clear_cannotGetUserData
+ * @brief               Test case of controller_screen_clear() where wl_resource_get_user_data() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_clear()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_clear_cannotGetUserData)
+{
+    controller_screen_clear(nullptr, nullptr);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_clear_success
+ * @brief               Test case of controller_screen_clear() where wl_resource_get_user_data() success, return an object
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_screen_clear()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() not be called
+ *                         +# screen_set_render_order() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_clear_success)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviScreen, 1);
+
+    controller_screen_clear(nullptr, nullptr);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 0);
+    ASSERT_EQ(screen_set_render_order_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_add_layer_nullScreen
+ * @brief               Test case of controller_screen_add_layer() where wl_resource_get_user_data() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_add_layer()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_add_layer_nullScreen)
+{
```

Comment:
- explicit wl_resource_get_user_data return null
- verify the post IVI_WM_SCREEN_ERROR_NO_SCREEN event.